core/orchestrator.go

```diff
@@ -316,37 +316,41 @@ func (n *LivepeerNode) transcodeSeg(config transcodeConfig, seg *stream.HLSSegme
 	start := time.Now()
 	tData, err := transcoder.Transcode(url, md.Profiles)
 	if err != nil {
-		glog.Errorf("Error transcoding seg: %v - %v", seg.Name, err)
+		glog.Errorf("Error transcoding manifest=%s segNo=%d segName=%s - %v", string(md.ManifestID), seg.SeqNo, seg.Name, err)
 		return terr(err)
 	}
 	// transcodeEnd := time.Now().UTC()
 	if len(tData) != len(md.Profiles) {
-		glog.Errorf("Did not receive the correct number of transcoded segments; got %v expected %v", len(tData), len(md.Profiles))
+		glog.Errorf("Did not receive the correct number of transcoded segments; got %v expected %v manifest=%s seqNo=%d", len(tData),
+			len(md.Profiles), string(md.ManifestID), seg.SeqNo)
 		return terr(fmt.Errorf("MismatchedSegments"))
 	}
+
 	took := time.Since(start)
 	tProfileData := make(map[ffmpeg.VideoProfile][]byte, 0)
-	glog.V(common.DEBUG).Infof("Transcoding of segment %v took %v", seg.SeqNo, took)
-	if isLocal && monitor.Enabled {
-		monitor.LogSegmentTranscodeEnded(seg.SeqNo, string(md.ManifestID), took,
-			common.ProfilesNames(md.Profiles))
-	}
 
 	// Prepare the result object
 	var tr TranscodeResult
 	segHashes := make([][]byte, len(tData))
 
-	for i, _ := range md.Profiles {
-		if tData[i] == nil {
-			glog.Errorf("Cannot find transcoded segment for %v", seg.SeqNo)
-			continue
+	for i := range md.Profiles {
+		if tData[i] == nil || len(tData[i]) < 25 {
```

Comment:
Where does the 25 come from here?

Reply:
Size of empty frame used in unit tests. If you increase this this constant here, you'll see tests fails.